lib/json/schema_builder/entity.rb

```diff
@@ -138,13 +139,16 @@ def _reset_fragments
       end
 
       def extract_types
-        any_of(null) if @nullable
-        if any_of.present?
-          everything_else = schema.data.reject { |k, v| k == "anyOf" }
-          return unless everything_else.present?
-          schema.data.select! { |k, v| k == "anyOf" }
-          schema.data["anyOf"].unshift everything_else
+        return unless any_of.present?
+        everything_else = schema.data.reject { |k, v| k == "anyOf" }
+        return unless everything_else.present?
+        schema.data.select! { |k, v| k == "anyOf" }
+        if initial = schema.data['anyOf'].find { |opt| opt.as_json.try(:[], 'type') == 'object' }
```

Comment:
I assume that, when some is extending this, they intend to extend the object, not the null, so I extract the initial definition and extend it here. This is really ugly right now though and probably won't work in some cases. Happy for any comments here.